Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Description: Makefile hardening, link shared libraries not hardcoded static ones
----------------------------------------------------------------------------------
diff -Nru CalculiX/ccx_2.20/src/Makefile CalculiX.1/ccx_2.20/src/Makefile
--- CalculiX/ccx_2.20/src/Makefile	2022-07-31 17:06:31.000000000 +0100
+++ CalculiX.1/ccx_2.20/src/Makefile	2022-12-06 16:24:27.000000000 +0000
@@ -1,9 +1,9 @@
 
-CFLAGS = -Wall -O2  -I ../../../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DNETWORKOUT
-FFLAGS = -Wall -O2
+CFLAGS += -Wall -O2  -I$(SPOOLES_INC_DIR) -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DNETWORKOUT
+FFLAGS += -Wall -O2
 
-CC=cc
-FC=gfortran
+CC?=cc
+FC?=gfortran
 
 .c.o :
 	$(CC) $(CFLAGS) -c $<
@@ -18,15 +18,10 @@
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../../../SPOOLES.2.2
+LIBS = -lspooles -lpthread -larpack -lflexiblas -lm -lc
 
-LIBS = \
-       $(DIR)/spooles.a \
-	../../../ARPACK/libarpack_INTEL.a \
-       -lpthread -lm -lc
-
-ccx_2.20: $(OCCXMAIN) ccx_2.20.a  $(LIBS)
-	./date.pl; $(CC) $(CFLAGS) -c ccx_2.20.c; $(FC)  -Wall -O2 -o $@ $(OCCXMAIN) ccx_2.20.a $(LIBS) -fopenmp
+ccx_2.20: $(OCCXMAIN) ccx_2.20.a
+	./date.pl; $(CC) $(CFLAGS) -c ccx_2.20.c; $(FC)  -Wall -O2 -o $@ $(OCCXMAIN) ccx_2.20.a $(LIBS) -fopenmp $(LDFLAGS)
 
 ccx_2.20.a: $(OCCXF) $(OCCXC)
 	ar vr $@ $?
